tiny_dnn/activations/softplus_layer.h

```diff
@@ -0,0 +1,49 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class softplus_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "softplus-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) override {
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      if (x[j] >= float_t(30)) {
```

Comment:
No this is not a parameter.
This is an approximation:
```
>> log(1+exp(30))
30.000000000000092
```
This approximation has been used by the theano guys and limits a lot of computation and favours numerical stability. 
```
exp(30) = 1.0686475e+13
```